component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -356,7 +356,7 @@ private void updateMobileNumberForUsername(AuthenticationContext context, HttpSe
                 log.debug("Updating mobile number for user : " + username);
             }
             Map<String, String> attributes = new HashMap<>();
-            attributes.put(SMSOTPConstants.MOBILE_CLAIM, request.getParameter(SMSOTPConstants.MOBILE_NUMBER));
+            attributes.put(SMSOTPConstants.MOBILE_CLAIM, String.valueOf(context.getProperty(SMSOTPConstants.REQUEST_USER_MOBILE)));
```

Comment:
does String.valueOf handle the case where context.getProperty can be null?

Reply:
updateMobileNumberForUsername is being called only if the context value is not null.